app/models/admin/shelter.rb

```diff
@@ -0,0 +1,14 @@
+class Admin::Shelter < ApplicationRecord
+  validates :name, presence: true
+  validates :rank, presence: true, numericality: true
+  validates :city, presence: true
+
+  has_many :pets, dependent: :destroy
+  
+  def find_shelter_name(shelter_id)
```

Comment:
As mentioned in the controller, it's not necessary to add an additional Admin::Shelter model but to just build this query in the existing Shelter model. When we add the the admin/shelter page, we are not adding additional data but just seeing the data displayed differently